event-store-plans.md

```diff
@@ -0,0 +1,174 @@
+aggregation data as part of the outgoing message.
+
+More generically, we need a good way to emit new events that will be processed as appends within the same daemon transaction. This might be a little tricky and will
+require a spike.
+
+**Gotta do this in a way such that users can opt into whether the new events should be emitted during projection rebuilds**
+
+
+## Sharding the Event Store tables
+
+This is a [long standing issue](https://github.com/JasperFx/marten/issues/770) that we didn't get around to in the huge V4 release.
+
+Some notes:
+
+* My personal proposal is to use native PostgreSQL sharding on the `is_archived` column to effect a "hot storage, cold storage" approach. This should be paired with
+  advice to users to be more aggressive about archiving event streams that are obsolete/stale/not active as a way to improve throughput over all. The daemon and every
+  other place where we query events automatically uses `is_archived = false` filters anyway
+* We probably shouldn't be sharding on anything that would impact the async daemon's "high water detection" that uses the event sequence. In other words, we
+  want to make it as easy as possible to walk the sequence number values within a single table
+* **Maybe** it would be valuable to shard the active events and event streams on the event sequence number ranges. That would be beneficial to the async daemon, if
+  harmful to live aggregations
+* This is going to be an ugly migration, which is why is was cut from V4. If you apply this after the fact, you'll have to have some down time. You'll have to copy the existing mt_streams & mt_events tables off to the side, then drop both tables, create the new table partitions and the virtual table that points to the partitions, then copy in all the events back into the virtual table so postgres can sort the actual records around
+
+
+## Sharding the Document Tables
+
+Same migration issues as the event tables, but there's just more possibilities here. We'll need a significant investment in Weasel for the migrations. There's
+some nascent partition support in Weasel already. Need some easy ways for users to create their own partitioning strategies
+
+
+## IEventSession interface/service
+
+What if we had a new interface specifically for folks who really only use the event sourcing at different areas of the code that got you straight to 
+event sourcing behavior like:
+
+```csharp
+public interface IEventSession : IEventStore
+{
+    Task SaveChangesAsync(CancellationToken token = default);
+}
+```
+
+It'd be the `DocumentSession`, but with a more event store friendly interface so you don't have to do `session.Events.******` on everything. 
+
+
+## Document by document type identity map behavior
+
+Today, document sessions either have the identity map behavior for all document types or no identity map for all types. I think especially around the 
```

Comment:
I think that maybe it'd be better to treat that as an internal option for projection usage. Making that public would make the number of permutations even higher.

Reply:
We've just got to have the capability is all. It's actually not going to be that big of a deal mechanically. Really just an enabler step